db-service/lib/cqn2sql.js

```diff
@@ -1484,9 +1484,9 @@ Readable.prototype[require('node:util').inspect.custom] = Readable.prototype.toJ
 
 const ObjectKeys = o => (o && [...ObjectKeys(o.__proto__), ...Object.keys(o)]) || []
 const _managed = {
-  '$user.id': '$user.id',
-  $user: '$user.id',
-  $now: '$now',
+  '$user.id': cds.requires.kinds.hana.impl.startsWith('@cap-js') ?  'APPLICATIONUSER' : '$user.id',
```

Comment:
we don't want to have hana specifics in the generic db-service implementation.
I think we should solve it similar to postgres:

https://github.com/cap-js/cds-dbs/blob/bd1f52f67fb4709dce3a27fea8856cb9b875da6b/postgres/lib/PostgresService.js#L68-L81